### PR TITLE
Remove logGainStandardResource and logGainProduction, logging is part of addResource and addProduction.

### PR DESCRIPTION
**Pull request by kberg**
The only sad part is that logGainStandardResource's text is better English.

### CODE REVIEW

File: src/cards/base/MartianRails.ts [hunk 1]
```diff
@@ -36,9 +36,8 @@ export class MartianRails extends Card implements IActionCard, IProjectCard {
     return player.energy >= 1;
   }
   public action(player: Player) {
-    player.energy--;
+    player.addResource(Resources.ENERGY, -1);
```
Comment (bafolts) [inline src/cards/base/MartianRails.ts:39]: This should probably be `deductResource`.

File: src/cards/base/PowerInfrastructure.ts [hunk 2]
```diff
@@ -40,9 +39,8 @@ export class PowerInfrastructure extends Card implements IActionCard, IProjectCa
       'Select amount of energy to spend',
       'Spend energy',
       (amount: number) => {
-        player.energy -= amount;
-        player.megaCredits += amount;
-        LogHelper.logGainStandardResource(player, Resources.MEGACREDITS, amount);
+        player.addResource(Resources.ENERGY, -amount);
```
Comment (bafolts) [inline src/cards/base/PowerInfrastructure.ts:42]: This should use `deductResource`.

**Review by bafolts — APPROVED** [review]

File: src/cards/base/MartianRails.ts (hunk 1 above)
Comment (kberg) [reply src/cards/base/MartianRails.ts:39]: Yep.

File: src/cards/base/PowerInfrastructure.ts (hunk 2 above)
Comment (kberg) [reply src/cards/base/PowerInfrastructure.ts:42]: :+1:

**Pull request merged**